### PR TITLE
Makes sounds with a low enough volume only audible within line of sight

### PR DESCRIPTION
**Pull request by skull132**
It has been posited that it doesn't make sense to hear certain sounds through walls. So, I proceeded to go and change that.

All sounds with a volume of 50 or less will now be played only to the "hearers" list. This is basically people within line of sight. I also rewrote the lower levels of the sound API with this. The code is now more modular, while retaining the same main API entry point `playsound`.

This needs a test merge to see how badly I broke shit.

### CODE REVIEW

**Comment by skull132** [comment]
!review

**Comment by ParadoxSpace** [comment]
Is it possible to list what sounds would now newly be invisible?

**Comment by skull132** [comment]
image
Yeah no.

**Comment by ParadoxSpace** [comment]
wew nevermind. sounds good tho.

**Comment by skull132** [comment]
Though tbh. _Most_ of them at this rate. That's kinda why I want this test merged, to see if I should only account for sounds that are at volume levels less than 50.

**Comment by alsoandanswer** [comment]
Conveniently, people don't have to meme my dropsounds or arcade noises anymore...

**Comment by KingOfThePing** [comment]
Finally, I don't have to hear security dropping their pants while ERPing in maintenance anymore.

File: code/controllers/subsystems/explosives.dm [hunk 1]
```diff
@@ -150,11 +149,11 @@ var/datum/controller/subsystem/explosives/SSexplosives
...
 						// If inside the blast radius + world.view - 2
 						if(dist <= closedist)
-							M.playsound_local(epicenter, get_sfx("explosion"), min(100, volume), 1, frequency, falloff = 5) // get_sfx() is so that everyone gets the same sound
+							M.playsound_simple(epicenter, get_sfx("explosion"), min(100, volume), use_random_freq = TRUE, falloff = 5)
 							//You hear a far explosion if you're outside the blast radius. Small bombs shouldn't be heard all over the station.
 
 						else
-							volume = M.playsound_local(epicenter, 'sound/effects/explosionfar.ogg', volume, 1, frequency, usepressure = 0, falloff = 1000)
+							volume = M.playsound_simple(epicenter, 'sound/effects/explosionfar.ogg', volume, use_random_freq = TRUE, falloff = 1000, use_pressure = FALSE)
 							//Playsound local will return the final volume the sound is actually played at
```
Comment (MarinaGryphon) [inline code/controllers/subsystems/explosives.dm]: Does `playsound_simple` have this functionality? Was it just a rename?

**Pull request merged**